### PR TITLE
Production parameters

### PR DESCRIPTION
**Pull request by Thom1729**
Formally define parameterized productions as a “family” of productions. Inspired by the [ECMAScript spec](https://tc39.es/ecma262/#sec-grammar-notation).

As a result of this change, the production rules no longer contain variables or conditional logic. Productions that once had conditional logic:

```
[#] nb-double-text(n,c) ::=
  c == FLOW-OUT  => nb-double-multi-line(n)
  c == FLOW-IN   => nb-double-multi-line(n)
  c == BLOCK-KEY => nb-double-one-line
  c == FLOW-KEY  => nb-double-one-line
```

Are now defined as multiple productions:

```
[#] s-line-prefix(n,BLOCK-OUT) ::= s-block-line-prefix(n)
    s-line-prefix(n,BLOCK-IN)  ::= s-block-line-prefix(n)
    s-line-prefix(n,FLOW-OUT)  ::= s-flow-line-prefix(n)
    s-line-prefix(n,FLOW-IN)   ::= s-flow-line-prefix(n)
```

Because the old `auto-detect()` thing in `c-indentation-indicator` was never defined, this PR removes it. To ensure that block scalar content indentation is well defined, I've expanded and reworded section 8.1.1.1. Block Indentation Indicator. (This could probably be done on its own, but I think it's a prerequisite for this PR.)

Comments welcome.

### CODE REVIEW

**Comment by Thom1729** [comment]
https://spec.yaml.io/pr-182

File: spec/1.2.2/spec.md [hunk 1]
```diff
@@ -4827,33 +4865,35 @@ variables.
...
+* [c-b-block-header(t)] <!-- _#_Empty_header↓ 01_#_Indentation_indicator↓ +_#_Chomping_indicator↓ 01-_#_Both_indicators↓ -->
 
 
 #### #. Block Indentation Indicator
 
-Typically, the [indentation] level of a [block scalar] is detected from its
-first non-[empty] line.
-It is an error for any of the leading [empty lines] to contain more [spaces]
-than the first non-[empty line].
+Every block scalar has a _content indentation level_.
```
Comment (ingydotnet) [inline spec/1.2.2/spec.md:4873]: Not sure this is a complete translation.

Comment (ingydotnet) [reply spec/1.2.2/spec.md:4873]: nm

**Review by ingydotnet — APPROVED** [review]
This is a great start to simplifying the grammar productions.

**Review by pantoniou — APPROVED** [review]
A bit over my head, but I *think* I get the gist of it...

**Pull request merged**